docs/research-software/mitgcm/mitgcm.md

```diff
@@ -99,3 +104,163 @@ each for up to one hour.
+
+The ECCO version 4 state estimate (ECCOv4-r4) is an observationally-constrained numerical solution produced by the ECCO group at JPL. If you would like to reproduce the state estimate on ARCHER2 in order to create customised runs and experiments, follow the instructions below. They have been slightly modified from the JPL instructions for ARCHER2. 
+
+For more information, see the ECCOv4-r4 website <https://ecco-group.org/products-ECCO-V4r4.htm>
+
+### Get the ECCOv4-r4 source code
+
+First, create a working directory, perhaps MYECCO. Navigate into this working directory:
+
+    mkdir MYECCO
+    cd MYECCO
+    
+In order to reproduce ECCOv4-r4, we need a specific checkpoint of the MITgcm source code. 
+
+    git clone https://github.com/MITgcm/MITgcm.git -b checkpoint66g
+    
+Next, get the ECCOv4-r4 specific code from GitHub:
+
+    cd MITgcm
+    mkdir -p ECCOV4/release4
+    cd ECCOV4/release4
+    git clone https://github.com/ECCO-GROUP/ECCO-v4-Configurations.git
+    mv ECCO-v4-Configurations/ECCOv4\ Release\ 4/code .
+    rm -rf ECCO-v4-Configurations
+    
+### Get the ECCOv4-r4 forcing files
+
+The surface forcing and other input files that are too large to be stored on GitHub are available via NASA data servers. In total, these files are about 200 GB in size. You must register for an Earthdata account and connect to a WebDAV server in order to access these files. For more detailed instructions, read the help page <https://ecco.jpl.nasa.gov/drive/help>.
+
+First, apply for an Earthdata account: <https://urs.earthdata.nasa.gov/users/new>
+
+Next, acquire your WebDAV credentials: <https://ecco.jpl.nasa.gov/drive> (second box from the top)
+
+Now, you can use wget to download the required forcing and input files:
+
+    wget -r --no-parent --user YOURUSERNAME --ask-password 
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_forcing
+
+    wget -r --no-parent --user YOURUSERNAME --ask-password 
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_init
+    
+    wget -r --no-parent --user YOURUSERNAME --ask-password
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_ecco
+
+### Compiling and running ECCOv4-r4
+
+Follow the instructions in the above section on building MITgcm on ARCHER2. These steps also work for ECCOv4-r4. You will need to create a build directory first:
+
+    cd MITgcm/ECCOV4/release4
+    mkdir build
+    cd build
+    
+Once you have compiled the model, you will have the mitgcmuv executable. 
+
+#### Issue with DOS formatting and end-of-namelist characters
+
+As of 16 December 2020, some of the ECCOv4-r4 files downloaded from the ECCO GitHub repository appear to be DOS formatted and/or have inconsistent end-of-namelist characters. This causes end-of-file runtime errors. The simplest solution appears to be running the dos2unix command in the namelist directory:
+
+    cd ../input_init/NAMELIST/
+    dos2unix data*
```

Comment:
I cannot see the `dos2unix` command on ARCHER2. Did you load a module to get access to this tool?

Reply:
Ah, good catch. I ran that fix on ARCHER, but you're right that it's not (yet?) available on ARCHER2. I'll fix that statement.